### PR TITLE
Merge Live into Test

### PR DESCRIPTION
**Pull request by magicmilo**
#### Work done
- Merged Live branch into Test branch

#### Tests
Does this PR include tests for the changes introduced?
- [ ] Yes
- [ ] No, I found it difficult to test
- [x] No, they are not required for this change

### CODE REVIEW

File: server/models/postcodes.js [hunk 1]
```diff
@@ -30,33 +105,90 @@ module.exports = function (sequelize, DataTypes) {
   );
 
   // There is no PK key on the postcodes table
+  // TODO this is not my comment above. We now have pk uuid.!
```
Comment (sbt92) [inline server/models/postcodes.js:108]: Should we clear these comments out if they're offering no value anymore?

File: server/models/postcodes.js [hunk 2]
```diff
@@ -30,33 +105,90 @@ module.exports = function (sequelize, DataTypes) {
...
-      Latitude: pcodeData.latitude,
-      Longitude: pcodeData.longitude,
-    });
+      // Some records with this postcode are not full so we delete all with this postcode
+      // if getAddressAPI returns results
+      if (foundPostcodes) {
+        await this.destroy({ where: { postcode: postcode } });
+      }
+
+      console.log(`New or updated postcodes data for: ${getAddressAPIResults.postcode}`);
```
Comment (sbt92) [inline server/models/postcodes.js:161]: Do we want this log?

File: server/routes/establishments/bulkUpload/lock.js [hunk 3]
```diff
@@ -141,6 +141,8 @@ const lockStatusGet = async (req, res) => {
 
   const currentLockState = await lockStatus(establishmentId);
 
+  console.log(`DBG currentLockState ${currentLockState}`);
```
Comment (sbt92) [inline server/routes/establishments/bulkUpload/lock.js]: Do we still need this?

File: src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts [hunk 4]
```diff
@@ -91,22 +95,24 @@ describe('StandAloneAccountComponent', () => {
   describe('Tabs', () => {
     it('should show all tabs when all permissions are on the establishment', async () => {
       const { component } = await setup();
-
-      expect(component.tabs).toEqual([homeTab, workplaceTab, staffRecordsTab, tAndQTab, benchmarksTab]);
+       expect(component.tabs).toEqual([homeTab, workplaceTab, staffRecordsTab, tAndQTab, benchmarksTab]);
+     // expect(component.tabs).toEqual([homeTab, workplaceTab, staffRecordsTab, tAndQTab]);
```
Comment (sbt92) [inline src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts:99]: Is this still needed?

File: src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts [hunk 5]
```diff
@@ -91,22 +95,24 @@ describe('StandAloneAccountComponent', () => {
...
     });
 
     it('should show not show the workplace tab when canViewEstablisment permission is not on the establishment', async () => {
       const permissions = ['canViewBenchmarks', 'canViewListOfUsers', 'canViewListOfWorkers'];
       const { component } = await setup(true, permissions);
 
       expect(component.tabs).toEqual([homeTab, staffRecordsTab, tAndQTab, benchmarksTab]);
+     // expect(component.tabs).toEqual([homeTab, staffRecordsTab, tAndQTab]);
```
Comment (sbt92) [inline src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts:107]: Is this still needed?

File: src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts [hunk 6]
```diff
@@ -91,22 +95,24 @@ describe('StandAloneAccountComponent', () => {
...
     });
 
     it('should show not show the staff-records or tAndQ tabs when canViewListOfWorkers permission is not on the establishment', async () => {
       const permissions = ['canViewBenchmarks', 'canViewListOfUsers', 'canViewEstablishment'];
       const { component } = await setup(true, permissions);
 
       expect(component.tabs).toEqual([homeTab, workplaceTab, benchmarksTab]);
+     // expect(component.tabs).toEqual([homeTab, workplaceTab]);
```
Comment (sbt92) [inline src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts:115]: Is this still needed?

**Review by sbt92 — APPROVED** [review]
Few suggestions that may be worth addressing due to the low effort to implement them but don't let them stop you from merging if you feel there is value in them being present in the codebase.

File: server/routes/establishments/bulkUpload/lock.js (hunk 3 above)
Comment (magicmilo) [reply server/routes/establishments/bulkUpload/lock.js]: Removed

File: server/models/postcodes.js (hunk 2 above)
Comment (magicmilo) [reply server/models/postcodes.js:161]: Useful

File: src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts (hunk 6 above)
Comment (magicmilo) [reply src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts:115]: Needs to be reintroduced when we go through all the tests

File: src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts (hunk 4 above)
Comment (magicmilo) [reply src/app/core/components/standAloneAccount/standAloneAccount.component.spec.ts:99]: Needs to be reintroduced when we go through all the tests

**Pull request merged**